spec/javascripts/unit/save-location.spec.js

```diff
@@ -1,184 +1,181 @@
 describe('Saving bank holiday locations', function () {
-  var $locations
-  var html = '<section class="js-save-nation" data-module="save-bank-holiday-nation" data-nation="England_and_Wales" data-nation-matches=\'["England","Wales"]\' data-save-description="Do you want to save # as your location for bank holidays?" data-save-button-aria-label="Save # as your location for bank holidays" data-save-button-text="Save" data-undo-description="You\'ve saved # as your location for bank holidays" data-undo-button-aria-label="Remove # as your saved location for bank holidays" data-undo-button-text="Undo">' +
-    '<p class="js-nation-description" aria-live="polite">Do you want to save # as your location for bank holidays?</p>' +
-    '<button class="js-nation-button" type="submit" aria-label="Save # as your location for bank holidays">Save</button>' +
-    '<p class="js-nation-link">' +
-      '<a href="#" class="govuk-link">We\'ll add a cookie to your device</a>' +
-    '</p>' +
-  '</section>'
+  var element
 
   beforeEach(function () {
     window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+    element = document.createElement('section')
+    element.classList.add('js-save-nation')
+    element.setAttribute('data-nation', 'England_and_Wales')
+    element.setAttribute('data-nation-matches', '["England","Wales"]')
+    element.setAttribute('data-save-description', 'Do you want to save # as your location for bank holidays?')
+    element.setAttribute('data-save-button-aria-label', 'Save # as your location for bank holidays')
+    element.setAttribute('data-save-button-text', 'Save')
+    element.setAttribute('data-undo-description', 'You\'ve saved # as your location for bank holidays')
+    element.setAttribute('data-undo-button-aria-label', 'Remove # as your saved location for bank holidays')
+    element.setAttribute('data-undo-button-text', 'Undo')
+    element.innerHTML =
+      '<p class="js-nation-description" aria-live="polite">Do you want to save # as your location for bank holidays?</p>' +
+      '<button class="js-nation-button" type="submit" aria-label="Save # as your location for bank holidays">Save</button>' +
+      '<p class="js-nation-link">' +
+        '<a href="#" class="govuk-link">We\'ll add a cookie to your device</a>' +
+      '</p>'
```

Comment:
To be honest, I'm not sure this makes it any easier to read. I think a better formatted html block might have been nicer.

Reply:
That's fair. I didn't make the change to improve readability the goal was removing jQuery and, well, jQuery is inherently more readable than the DOM APIs 😆 .

I can see how this ended up rather clunky as there is so many data attributes. I've switched the approach of this to another technique that should hopefully resolve your concern.

Reply:
The `div.innerHTML` is much nicer thanks!

For me the benefit of the html block is that it's easier to see which bits of the html match which element attributes. I